Update doc/persistence.md (#641)
jenkinsci/workflow-cps-plugin: doc/persistence.md

Some basic rules:

> 1. If the FlowNodeStorage is corrupt, incomplete, or un-persisted, all manner of check will break loose
>     - In terms of Pipeline execution, the impact is like the Resonance Cascade from the Half-Life games

Reviewer: This isn't a typo, it's a euphemism for "hell". That said, maybe better to reword things for clarity and to avoid cultural references.

```suggestion
1. If the FlowNodeStorage is corrupt, incomplete, or un-persisted, various things will break
```
